### PR TITLE
add id to KinClient, for storing a different data set per id

### PR DESCRIPTION
**Pull request by yosriz**
* add id to KinClient ctor
* add tests

### CODE REVIEW

File: kin-core/src/androidTest/java/kin/core/KinClientIntegrationTest.java [hunk 1]
```diff
@@ -25,15 +27,30 @@
...
+    private KinClient kinClient2;
 
     @Rule
     public ExpectedException expectedEx = ExpectedException.none();
 
     @Before
     public void setup() {
         serviceProvider = new ServiceProvider(TEST_NETWORK_URL, TEST_NETWORK_ID);
-        kinClient = new KinClient(InstrumentationRegistry.getTargetContext(), serviceProvider);
+        kinClient = createNewKinClient("test");
```
Comment (OrenZak) [inline kin-core/src/androidTest/java/kin/core/KinClientIntegrationTest.java]: use `test` and `test2` as constants please

File: kin-core/src/androidTest/java/kin/core/KinClientIntegrationTest.java [hunk 2]
```diff
@@ -255,4 +271,37 @@ public void getServiceProvider() throws Exception {
...
+    public void testMultipleKinAccounts_AddAccount_DifferentAccounts() throws Exception {
+        KinAccount account = kinClient.addAccount();
+
+        kinClient2 = createNewKinClient("test2");
+        assertThat(kinClient2.getAccount(0), nullValue());
+
+        KinAccount account2 = kinClient2.addAccount();
+
+        kinClient = createNewKinClient("test");
+        assertThat(kinClient.getAccount(0), equalTo(account));
```
Comment (OrenZak) [inline kin-core/src/androidTest/java/kin/core/KinClientIntegrationTest.java]: can you add the same check also for kinClient2, just to be on the safe side :)

File: kin-core/src/main/java/kin/core/KinClient.java [hunk 3]
```diff
@@ -30,12 +30,13 @@
      * KinClient is an account manager for a {@link KinAccount}.
      *
      * @param context the android application context
-     * @param provider the service provider provides blockchain network parameters
+     * @param id unique id for identifying this client, different ids will store a different accounts
+     * @param provider the service provider - provides blockchain network parameters
      */
-    public KinClient(@NonNull Context context, @NonNull ServiceProvider provider) {
+    public KinClient(@NonNull Context context, @NonNull String id, @NonNull ServiceProvider provider) {
```
Comment (OrenZak) [inline kin-core/src/main/java/kin/core/KinClient.java]: Does `id` should optional and use default? 
It's not a must for every digital service right?

File: kin-core/src/main/java/kin/core/KinClient.java [hunk 4]
```diff
@@ -117,6 +120,7 @@ public boolean hasAccount() {
     /**
      * Returns the number of existing accounts
      */
+    @SuppressWarnings("WeakerAccess")
```
Comment (OrenZak) [inline kin-core/src/main/java/kin/core/KinClient.java:132]: why did you add this now ?

**Review by OrenZak — APPROVED** [review]
👍

File: kin-core/src/main/java/kin/core/KinClient.java (hunk 3 above)
Comment (yosriz) [reply kin-core/src/main/java/kin/core/KinClient.java]: rename to `storageKey`

File: kin-core/src/main/java/kin/core/KinClient.java (hunk 4 above)
Comment (yosriz) [reply kin-core/src/main/java/kin/core/KinClient.java:132]: those are old false warnings that I got tired of (those methods are public intentionally though are not used in sample app)
